sphinx/addnodes.py

```diff
@@ -91,7 +96,7 @@ class desc_signature(nodes.Part, nodes.Inline, nodes.TextElement):
     """
 
 
-class desc_signature_line(nodes.Part, nodes.Inline, nodes.TextElement):
+class desc_signature_line(nodes.Part, nodes.Inline, nodes.FixedTextElement):
```

Comment:
I think leaving TextElement should now be ok. Indeed the docutils.transforms.universal code contains:

```
# nested TextElements are not "block-level" elements:
            if isinstance(node.parent, nodes.TextElement):
                continue
```

So `desc_signature_line` is skipped as child of `desc_signature`. Only the latter is walked for its Text nodes. But now that you assign FixedTextElement class to `desc_signature` it is skipped earlier:
```
            # skip preformatted text blocks and special elements:
            if isinstance(node, (nodes.FixedTextElement, nodes.Special)):
                continue
```